tests/ci/common_posix_setup.sh

```diff
@@ -0,0 +1,244 @@
+  # For grep's basic regular expression language the meta-characters (e.g. "?",
+  # "|", etc.) are interpreted as literal characters. To keep their
+  # meta-character semantics, they must be escaped with "\".
+  # Deciphering the pattern "^_\?\(bignum\|curve25519_x25519\)":
+  #  * "^": anchor at start of line.
+  #  * "_\?": might contain underscore.
+  #  * "\(bignum\|curve25519_x25519\)": match string of either "bignum" or "curve25519_x25519".
+  # Recall that the option "-v" reverse the pattern matching. So, we are really
+  # filtering out lines that contain either "bignum" or "curve25519_x25519".
+  cat "$BUILD_ROOT"/symbols_final_crypto.txt  "$BUILD_ROOT"/symbols_final_ssl.txt | grep -v -e '^_\?\(bignum\|curve25519_x25519\)' >  "$SRC_ROOT"/symbols_final.txt
+  # Now filter out every line that has the unique prefix $CUSTOM_PREFIX. If we
+  # have any lines left, then some symbol(s) weren't prefixed, unexpectedly.
+  if [ $(grep -c -v ${CUSTOM_PREFIX}  "$SRC_ROOT"/symbols_final.txt) -ne 0 ]; then
+    echo "Symbol(s) missing prefix!"
+    exit 1
+  fi
+}
+
+
+function build_prefix_and_test {
+  CUSTOM_PREFIX=aws_lc_1_1_0
+  run_build "$@"
+  generate_symbols_file
+  run_build "$@" "-DBORINGSSL_PREFIX=${CUSTOM_PREFIX}" "-DBORINGSSL_PREFIX_SYMBOLS=${SRC_ROOT}/symbols.txt"
+  verify_symbols_prefixed
+  run_cmake_custom_target 'run_tests'
+}
+
+function fips_build_and_test {
+  run_build "$@" -DFIPS=1
+  # Upon completion of the build process. The module’s status can be verified by 'tool/bssl isfips'.
+  # https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp3678.pdf
+  # FIPS mode is enabled when 'defined(BORINGSSL_FIPS) && !defined(OPENSSL_ASAN)'.
+  # https://github.com/aws/aws-lc/blob/220e266d4e415cf0101388b89a2bd855e0e4e203/crypto/fipsmodule/is_fips.c#L22
+  expect_fips_mode=1
+  for build_flag in "$@"
+  do
+    if [[ "${build_flag}" == '-DASAN=1' ]]; then
+      expect_fips_mode=0
+      break
+    fi
+  done
+  module_status=$("${BUILD_ROOT}/tool/bssl" isfips)
+  [[ "${expect_fips_mode}" == "${module_status}" ]] || { echo >&2 "FIPS Mode validation failed."; exit 1; }
+  # Run tests.
+  run_cmake_custom_target 'run_tests'
+  "${BUILD_ROOT}/util/fipstools/test_fips"
+}
+
+function build_and_test_valgrind {
+  if [[ $PLATFORM == "aarch64" ]]; then
+    run_build "$@" -DCMAKE_C_FLAGS="$VALGRIND_STATIC_CAP_FLAGS"
+    run_cmake_custom_target 'run_tests_valgrind'
+
+    # Disable all capabilities and run again
+    # (We don't use the env. variable OPENSSL_armcap because it is currently
+    #  restricted to the case of runtime discovery of capabilities
+    #  in cpu_aarch64_linux.c)
+    run_build "$@" -DCMAKE_C_FLAGS="-DOPENSSL_STATIC_ARMCAP"
+    run_cmake_custom_target 'run_tests_valgrind'
+  else
+    run_build "$@"
+    run_cmake_custom_target 'run_tests_valgrind'
+  fi
+}
+
+function build_and_test_ssl_runner_valgrind {
```

Comment:
note for later PR: a lot of the stuff in this file seems AWS-LC-specific. we should prune out what we can.

Reply:
Yup, you'll see that in the next PR.